test/jdk/java/util/zip/ZipInputStream/Zip64DataDescriptor.java

```diff
@@ -0,0 +1,99 @@
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @summary ZipInputStream should read 8-byte data descriptors if the LOC has
+ *   a ZIP64 extended information extra field
+ * @run testng Zip64DataDescriptor
+ */
+
+
+import org.testng.annotations.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.HexFormat;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+public class Zip64DataDescriptor {
+
+    @Test
+    public void shouldReadZip64Descriptor() throws IOException {
+
+        /**
+         * Structure of the ZIP64 file used below . Note the precense
+         * of a Zip64 extended information extra field and the
+         * Data Descriptor having 8-byte values for csize and size.
+         *
+         * ------  Local File Header  ------
+         * 000000  signature          0x04034b50
+         * 000004  version            45
+         * 000006  flags              0x0808
+         * 000008  method             8
+         * 000010  time               0x7e00         15:48
+         * 000012  date               0x564c         2023-02-12
+         * 000014  crc                0x00000000
+         * 000018  csize              -1
+         * 000022  size               -1
+         * 000026  nlen               5
+         * 000028  elen               20
+         * 000030  name               5 bytes        'entry'
+         * 000035  ext id             0x0001         Zip64 extended information extra field
+         * 000037  ext size           16
+         * 000039  z64 size           0
+         * 000047  z64 csize          0
+         * ------  File Data  ------
+         * 000055  ext data           7 bytes
+         * ------  Data Descriptor  ------
+         * 000062  signature          0x08074b50
+         * 000066  crc                0x3610a686
+         * 000070  csize              7
+         * 000078  size               5
+         * 000086  ...
+         */
+
+        String hex = """
+                504b03042d0008080800007e4c5600000000ffffffffffffffff05001400
+                656e7472790100100000000000000000000000000000000000cb48cdc9c9
+                0700504b070886a6103607000000000000000500000000000000504b0102
+                1400140008080800007e4c5686a61036ffffffffffffffff050020000900
+                ffff000000000000ffffffff656e74727901001c00050000000000000007
+                000000000000000000000000000000000000004120636f6d6d656e74504b
+                06062c000000000000002d002d0000000000000000000100000000000000
+                01000000000000005c000000000000005600000000000000504b06070000
+                0000b20000000000000001000000504b050600000000010001005c000000
+                560000000000""";
+
```

Comment:
This Zip was produced using my ZIP stream transformation library. This is not in the JDK, so probably not so useful to reference. Let me see if I can rework the test to use a Zip created using the `zip` streaming mode I described instead, that would allow a more independent reproduction.